src/main/java/org/tl/Hobbies.java

```diff
@@ -1,68 +1,173 @@
 package org.tl;
 
-import java.util.List;
-import java.util.Map;
+import java.io.*;
+import java.util.*;
 
 public class Hobbies {
 
+
     /**
      * Collect lines from the given file into a list.
      *
      * @param filePath The path to the file to read.
      * @return A list of lines from the file.
      */
     public List<String> createListFromFile(String filePath) {
-        return null;
+        List<String> lines = new ArrayList<>();
+        try {
+            File file = new File(filePath);
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                lines.add(line);
+            }
+
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            System.out.printf("Error occurred %s%n", e.getMessage());
+        }
+        return lines;
     }
 
+
     /**
      * Create a dictionary of people's hobbies from the given file.
      *
      * @param filePath The path to the file to read.
      * @return A map where each person's name is associated with a list of their hobbies.
      */
     public Map<String, List<String>> createDictionary(String filePath) {
-        return null;
+        Map<String, List<String>> result = new HashMap<>();
+        try {
+            FileReader fileReader = new FileReader(filePath);
+            Scanner scanner = new Scanner(fileReader);
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                String[] parts = line.split(":");
+                String name = parts[0].trim();
+                String[] hobbies = parts[1].split(",");
+                List<String> hobbyList = new ArrayList<>(Arrays.asList(hobbies));
+                result.put(name, hobbyList);
+            }
+            scanner.close();
+        } catch (FileNotFoundException ex) {
+            throw new RuntimeException("File not found:" + filePath);
+        }
+        return result;
     }
 
+
     /**
      * Find the person (or people) who have more hobbies than others.
      *
      * @param filePath The path to the file containing hobby data.
      * @return A list of people with the most hobbies.
      */
     public List<String> findPersonWithMostHobbies(String filePath) {
-        return null;
+        Map<String, List<String>> dictionary = createDictionary(filePath);
+        int maxHobbies = 0;
+        String result = null;
+        for (String name : dictionary.keySet()) {
+            int hobbiesCount = dictionary.get(name).size();
+            if (hobbiesCount > maxHobbies) {
+                maxHobbies = hobbiesCount;
+                result = name;
+            }
+        }
+        return Collections.singletonList(result);
```

Comment:
There might be a case, where there are people who have same amount of hobbies, so you need to return all those names and not only one